pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -665,15 +672,9 @@ protected synchronized boolean trySendMessagesToConsumers(ReadType readType, Lis
         long totalEntries = 0;
         int avgBatchSizePerMsg = remainingMessages > 0 ? Math.max(remainingMessages / entries.size(), 1) : 1;
 
-        int firstAvailableConsumerPermits, currentTotalAvailablePermits;
-        boolean dispatchMessage;
-        while (entriesToDispatch > 0) {
-            firstAvailableConsumerPermits = getFirstAvailableConsumerPermits();
-            currentTotalAvailablePermits = Math.max(totalAvailablePermits, firstAvailableConsumerPermits);
-            dispatchMessage = currentTotalAvailablePermits > 0 && firstAvailableConsumerPermits > 0;
```

Comment:
Why need to remove this check?

Reply:
Got it, thanks.

Reply:
Maybe the current total available permits should be the minimum value between `totalAvailablePermits` and `firstAvailableConsumerPermits`, right?

Reply:
Make sense, thanks.

Reply:
Make sense, thanks.